lib/internal/process/execution.js

```diff
@@ -238,10 +246,120 @@ function readStdin(callback) {
   });
 }
 
+/**
+ *
+ * Adds the TS message to the error stack
+ * At the 3rd line of the stack, the message is added
+ * @param {Error} originalStack the stack to decorate
+ * @param {string} newMessage the message to add to the error stack
+ * @returns {void}
+ */
+function decorateCJSErrorWithTSMessage(originalStack, newMessage) {
+  const lines = StringPrototypeSplit(originalStack, '\n');
+  ArrayPrototypeSplice(lines, 3, 0, newMessage);
+  return ArrayPrototypeJoin(lines, '\n');
```

Comment:
We can avoid the array allocation here
```suggestion
  let index;
  for (let i = 0; i < 3; i++) {
    index = StringPrototypeIndexOf(originalStack, '\n', index + 1);
  }
);
  return StringPrototypeSlice(originalStack, 0, index) + "\n" + newMessage + StringPrototypeSlice(originalStack, index);
```